README.md

```diff
@@ -3,5 +3,68 @@
 
 ## Testing
 
-Coverage report is available from
+### Preparation
+1. Clone the following repositories:
+    * https://github.com/osbuild/image-builder-frontend
+    * https://github.com/osbuild/image-builder
+    * https://github.com/RedHatInsights/insights-proxy
+
+2. Setting up the proxy
+
+    Choose a runner (podman or docker), and point the SPANDX_CONFIG variable to
+    profile/local-frontend-and-api-with-identity.js included in image-builder-frontend.
+
+    ```
+    # sudo insights-proxy/scripts/patch-etc-hosts.sh
+    # export RUNNER="podman"
+    # export SPANDX_CONFIG=$PATH_TO/image-builder-frontend/profiles/local-frontend-and-api-with-identity.js
+    # sudo -E insights-proxy/scripts/run.sh
+    ``` 
+
+3. Setting up osbuild-composer(-api)
+
+    Move to image-builder folder. The easiest way to do this is to call schutzbots/provision-composer.sh. This will install composer, generate the needed certs, and put the configuration in place.
+
+
+### Unit test
+
+* In image-builder folder, run all the unit tests:
+
+    ```
+    # go clean -testcache # Clean cache before rerun unit tests
+    # go test ./... # Recursively run all the unit tests
+    ```
+
+### Integration test
+
+1. Build image-builder docker image. 
+    
+    The easiest way is to call schutzbots/build.sh. It will build image-builder and image-builder-tests packages, install them into your testbed, and build an image-builder docker image
+    ```
+    # export WORKSPACE=.
+    # export JOB_NAME=
+    # sudo schutzbots/build.sh
+    ```
+
+2. Run integration test
+
+    The easist way is to call schutzbot/run_tests.sh. It will start a image-builder container and run image-builder-tests. Or alternatively, run the integration test manually:
+    
+    ```
+    # sudo podman run -d --pull=never --security-opt "label=disable" --net=host \
+     -e LISTEN_ADDRESS=localhost:8087 -e OSBUILD_URL=https://localhost:443/api/composer/v1 \
+     -e OSBUILD_CA_PATH=/etc/osbuild-composer/ca-crt.pem \
+     -e OSBUILD_CERT_PATH=/etc/osbuild-composer/client-crt.pem \
+     -e OSBUILD_KEY_PATH=/etc/osbuild-composer/client-key.pem \
+     -v /etc/osbuild-composer:/etc/osbuild-composer \
+     image-builder # Start image-builder container
+    # go test ./cmd/... -tags=integration # Run integration test
+    ```
+   
+### CI
+* https://osbuildci.cloud.paas.psi.redhat.com/me/my-views/view/all/job/osbuild/job/image-builder/job/main/
```

Comment:
This is an internal url, personally don't think it should be in the public readme.